nomt/src/beatree/mod.rs

```diff
@@ -196,7 +223,15 @@ impl Tree {
         let bbn_store;
         let io_handle;
         {
+            // Wait for all outstanding read transactions to conclude. This ensures they won't
+            // be invalidated by any destructive changes.
+            //
+            // Once this guard drops, it will be safe to initiate a new read transaction.
+            let read_tx_guard = read_transaction_latch.block_until_unique();
```

Comment:
I spent some time and I think I don't understand why this only blocks here but imediatelly released after we got the write. Why block then at all?

Reply:
It ensures that no new read transactions will be created until the write lock on `self.shared` is released in any case.

It would be fine to hold `read_tx_guard` until the end of this scope as well, of course. But not blocking would lead to a race, where we `block_until_unique` but then a new read transaction is created by another thread before we can acquire `shared.write()`.

Reply:
Sure, but why is it important to exclude the new readers only for the duration of this scope?